### PR TITLE
Cmake updates

### PR DESCRIPTION
**Pull request by amjames**
# Notable Changes
Due to [psi4-#554](https://github.com/psi4/psi4/issues/554) some changes are needed in the pasture cmake system as well. 

This PR adds those changes, and should be merged as soon as the PR to the main repo is merged. 

## Minor additions
 I also copied over the .gitignore from psi4/psi4 since this repo does not have one yet. 
 - [x] ready to go

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 if(ENABLE_transqt2 OR ENABLE_ccsort)
   if(NOT (ENABLE_ccsort AND ENABLE_transqt2))
     message(STATUS  "ccsort and transqt2 enabled since interdependent")
     set(ENABLE_ccsort "ON")
     set(ENABLE_transqt2 "ON")
   endif()
 endif()
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 find_package(psi4 1.0 REQUIRED)
+include(psi4PluginCache)
```
Comment (loriab) [inline CMakeLists.txt:12]: Hadn't thought of using the PluginCache like this directly (as opposed to `cmake -C <plugincache>`), but if it works both when called through psi4 build system and it build the repo independently, then sure.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
 # define the install path
-get_filename_component(psi4_lib_loc ${core_LIBRARY} DIRECTORY)
+get_filename_component(psi4_lib_loc ${psi4_LIBRARY} DIRECTORY)
 set(install_loc "${psi4_lib_loc}/driver/pasture")
```
Comment (loriab) [inline CMakeLists.txt:16]: Don't have to decide now, but what do you think of installing to `lib/${PYMOD_INSTALL_LIBDIR}/pasture/ccsort` instead of `lib/${PYMOD_INSTALL_LIBDIR}/psi4/driver/pasture/ccsort`? If `PYTHONPATH` can find python module psi4 (which clearly it can or psi4 couldn't work), then it can find pasture.

File: ccsort/CMakeLists.txt [hunk 3]
```diff
@@ -1,13 +1,7 @@
...
-target_link_libraries(ccsort ${core_LIBRARY})
-set_target_properties(ccsort PROPERTIES
-  PREFIX ""
-  COMPILE_FLAGS "-std=c++11")
-if(APPLE)
-  set_target_properties(ccsort PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
-endif()
+
+add_psi4_plugin(ccsort ${ccsort_sources})
+target_include_directories(ccsort SYSTEM PRIVATE ${psi4_INCLUDE_DIR})
```
Comment (loriab) [inline ccsort/CMakeLists.txt]: Out of curiosity, why is this line needed?

**Review by loriab — APPROVED** [review]
LGTM!

**Review by andysim — APPROVED** [review]
LGTM (although I'm far from an expert on these things, as I'm sure you can gather from my initial attempt at putting CMake into Psi4)

File: ccsort/CMakeLists.txt (hunk 3 above)
Comment (amjames) [reply ccsort/CMakeLists.txt]: I may be able to drop these lines. I had needed something like them previously (I think it lived one level up) so that I could find psi4 headers at compile time. The new machinery on psi4's side may have made this unnecessary. I will test this out

File: CMakeLists.txt (hunk 2 above)
Comment (amjames) [reply CMakeLists.txt:16]: I had it install in `/psi4/driver/pasture/..` so that multiple psi4-installs can live side by side with pasture installed or not. It was useful for me when I was setting this up to know that `pathA/bin/psi4` has pasture installed,  `pathB/bin/psi4`  does not. I intended for the pasture imports to be looking up internally for this reason. It may be that I did not do that correctly and it only seems to work because I don't usually add psi4 to my `PYTHONPATH`. 

If there is no "real-life" situation where it would matter if pasture was installed in all psi4 installs then maybe we should consider moving it outside of the psi4 build tree.

File: CMakeLists.txt (hunk 1 above)
Comment (amjames) [reply CMakeLists.txt:12]: It does work in both cases although I don't know what precedence it takes over command line arguments. This is basically using the settings in `psi4PluginCache.cmake` as the default settings for the project.

File: ccsort/CMakeLists.txt (hunk 3 above)
Comment (loriab) [reply ccsort/CMakeLists.txt]: The `psi4::core` target comes with psi4, pybind11, and python libraries attached, and the add plugin should be `target_link_libraries`-ing to that, so you may be able to drop it. But then Jet, too, had that set up another way. So, as you say, only experiment can decide. :-)

File: CMakeLists.txt (hunk 2 above)
Comment (loriab) [reply CMakeLists.txt:16]: I think you can still have the multiple psi4-installs side-by-side since each will have its own `lib`. And it'll still be in the psi4 build tree (since so far everything goes behind `CMAKE_INSTALL_PREFIX`). But again, this was more a planning query than something to address now.

Comment (amjames) [reply CMakeLists.txt:16]: Good point, I will leave it for now, and we can work that out + finalize it before the 1.1 release. I am not married to either way, I just thought it made more sense to install within psi4 since the module doesn't really have any use on its own that change is easy to do if we decide thats the way to go.

**Review by dgasmith — APPROVED** [review]
